pipelines/tekton/azureml-container-pipeline/kserve-download-model.yaml

```diff
@@ -12,13 +12,30 @@ spec:
     type: string
   - name: aws-bucket-name
     type: string
+  - name: fetch-github-model
+    type: string
+    default: "true"
+  - name: fetch-S3-model
+    type: string
+    default: "false"
+  - name: github-url
+    type: string
+    default: https://github.com/opendatahub-io/ai-edge/raw/3dd0242871371cf66af3f20e3a8924a4b5fc1658/pipelines/models/bike-rentals-auto-ml/model.pkl
   steps:
-  - name: download-model
+  - name: download-model-github
+    image: curlimages/curl:latest
+    script: |
+      if [ "$(params.fetch-github-model)" = "true" ]; then
+        curl -L $(params.github-url) -o $(workspaces.workspace.path)/$(params.model-name).pkl
+      fi  
```

Comment:
I believe git clone is preferable over curl in our use case: https://github.com/opendatahub-io/ai-edge/issues/56#issuecomment-1717325175

Reply:
@piotrpdev While using git clone, should we be git cloning https://github.com/opendatahub-io/ai-edge.git  or the mlflow repo? Only problem is that it would be very heavy and be cloning other models also. What do you think?

I was back to considering curling the whole folder just.

Reply:
Responded to this in DM's TL;DR:
* Clone https://github.com/opendatahub-io/ai-edge.git because that's where the generated model files are.
* Our repo isn't that heavy, make a shallow clone: https://stackoverflow.com/a/34657781/19020549
* Also see https://github.com/opendatahub-io/ai-edge/issues/56#issuecomment-1716393051